stock_picking_purchase_propagate/models/purchase.py

```diff
@@ -0,0 +1,24 @@
+# Copyright 2018 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api
+
+
+class PurchaseOrderLine(models.Model):
+
+    _inherit = 'purchase.order.line'
+
+    @api.multi
+    def _create_stock_moves(self, picking):
+        """ When creating the moves from a PO, propagate the procurement group
+            and quantity from the PO lines to the destination moves, and
+            reassign pickings.
+        """
+        moves = super(PurchaseOrderLine, self)._create_stock_moves(picking)
+        destination_moves_to_prop = moves.filtered(
+            lambda m: m.propagate).mapped('move_dest_ids').filtered(
```

Comment:
any chance to get this w/ single search?

Reply:
The only option I see is bypassing ORM with this :
```
        sql = """
        SELECT dest_mov.id
        FROM stock_move dest_mov
        INNER JOIN stock_move_move_rel rel ON rel.move_dest_id = dest_mov.id
        INNER JOIN stock_move orig_mov ON rel.move_orig_id = orig_mov.id
        WHERE orig_mov.propagate = TRUE 
        AND dest_mov.group_id IS NULL
        AND orig_mov.id IN %s;
        """
        self.env.cr.execute(sql, [tuple(moves.ids)])
        dest_moves_ids = [x[0] for x in self.env.cr.fetchall()]
        dest_moves = self.env['stock.move'].browse(dest_moves_ids)
```
Might still be better in terms of performance than using two filtered and a mapped functions. What do you think ?

Reply:
sounds good. When we get here everything should be already checked regarding permissions, no?

Reply:
As the `_create_stock_moves` already creates records on stock_move (implying user has write access), and the two tables used are `stock_move` and `stock_move_move_rel` (which is only a m2m relation table between two stock_move fields) for a read operation, IMO it should be ok.